### ISSUE TITLE
Kubectl log pod runs into i/o timeout without firewalls.

### ISSUE THREAD

**Issue by Kinjara** [Azure/AKS#1433]
**What happened**:
I use the following statement to create an AKS cluster - 
The principle used is [Monitoring Metric Publisher] and [Contributor] on an empty resource-group.

az aks create --resource-group $ResourceGroup --name $AKSClusterName --node-count 3 --zones 3 --max-pods 50 --network-plugin kubenet  --service-cidr 10.0.0.0/16  --dns-service-ip 10.0.0.10  --pod-cidr 10.244.0.0/16  --docker-bridge-address 172.17.0.1/16 --vnet-subnet-id $SubnetID --service-principal $AKSClientID  --client-secret $AKSClientSecret --kubernetes-version $AKS_Version --workspace-resource-id $OMSWorkspaceId --enable-addons monitoring  --generate-ssh-keys --api-server-authorized-ip-ranges $IPRange 

*known issue* the UDR doesnt get linked to existing VNETs as we dont allow the service-acocunt to become Network Contributor - we manually manage the UDR's post deployment.
Reason is twofold - we dont want the cluster id to have privileges and we need to route traffic via the firewall.

AKS comes up and reports success - when we check the internals - we cannot run Kubectl logs <pods> --namepsace kube-system  with other commando's working.

The result of the command is:
Error from server ... dail tcp ... i/o timeout.

**What you expected to happen**:
A fully functional Cluster.

**How to reproduce it (as minimally and precisely as possible)**:
Create the log-workspace, create a service principle and allow him [Monitoring Metric Publisher] and [Contributor] on an empty resource-group.
Take the hub-spoke model and route the traffic from the spoke to the central firewall in the UDR's.
pre-configure the UDR's to the cluster-nodes ( 1  / 3 should be correct or the node-pool is marked as failed)
 
publish AKS into the environment with kubenet and create a connection and ask logs from a pod, connect Azure Monitoring and see the error.

**Anything else we need to know?**:
During the execution we have two issues - the role linking of [Monitoring Metric Publisher] and [Network Contributor] runs into a Http403.
No blocked traffic on firewalls or additional configuration on the NSG's.

If we give the installation script - subscription owner - it works on a MSDN subscription.

**Environment**:
- Kubernetes version (use `kubectl version`): 1.15.5 - 1.16.4
- Size of cluster (how many worker nodes are in the cluster?): 3 nodes
- General description of workloads in the cluster (e.g. HTTP microservices, Java app, Ruby on Rails, machine learning, etc.): AKS Core - no secondary deployments - currently trying to fetch the log from kube-proxy.
- Others:

**Comment by seanmck** [comment]
Without network contributor, we can't attach the AKS-created route table to the vnet. I think this is the feature you need: #1424. We are investigating offering that capability.

Feel free to reopen if I misunderstood the ask.

**Comment by Kinjara** [comment]
Hi Sean,

*We manually added the UDR rules* - we can't work with traffic that is not routed to the central part for logging and audit-reasons so we used the standard Rout-table that only has a "Default 
- 0.0.0.0/0" -rule extra.
The VNET also has a reference to our own DNS-server instead of default so we can (in time) chatter with other resources that are deployed.

With that said - the pods, node-pool are in a healthy-state (Only the omsagent-pods is in a crash-loop every 10ish minutes) and i can run pods in the cluster - this cannot be achieved without the UDR - the problem is that i cannot query the logs or harvest them via the monitoring-addon.

So while #1424 will absolutely help to automate it; it is not the bane of my problem.
